fastestimator/backend/tensor_normalize.py

```diff
@@ -0,0 +1,123 @@
+import torch
+import numpy as np
+import tensorflow as tf
+
+
+from fastestimator.backend.reduce_mean import reduce_mean
+from fastestimator.backend.reduce_std import reduce_std
+from fastestimator.backend.to_tensor import to_tensor
+
+Tensor = TypeVar('Tensor', tf.Tensor, torch.Tensor, np.ndarray)
+
+
+def normalize(tensor: Tensor, mean, std, epsilon: float = 1e-7) -> Tensor:
+    """Compute the normalized value of a `tensor`.
+
+    This method can be used with Numpy data:
+    ```python
+    n = np.array([[[1.0, 2.0], [3.0, 4.0]], [[5.0, 6.0], [7.0, 8.0]]])
+    b = fe.backend.tensor_normalize(n, None, None)  # ([[[-1.52752516, -1.0910894 ], [-0.65465364, -0.21821788]], [[ 0.21821788,  0.65465364], [ 1.0910894 ,  1.52752516]]])
+    b = fe.backend.tensor_normalize(n, 4.5, 2.29128784747792)  # ([[[-1.52752516, -1.0910894 ], [-0.65465364, -0.21821788]], [[ 0.21821788,  0.65465364], [ 1.0910894 ,  1.52752516]]])
+    b = fe.backend.tensor_normalize(n, (4., 5.), (2.23606798, 2.23606798))  # [[[-1.34164073, -1.34164073], [-0.44721358, -0.44721358]], [[ 0.44721358,  0.44721358], [ 1.34164073,  1.34164073]]]
+    ```
+
+    This method can be used with TensorFlow tensors:
+    ```python
+    t = tf.constant([[[1.0, 2.0], [3.0, 4.0]], [[5.0, 6.0], [7.0, 8.0]]])
+    b = fe.backend.tensor_normalize(n, None, None)  # ([[[-1.52752516, -1.0910894 ], [-0.65465364, -0.21821788]], [[ 0.21821788,  0.65465364], [ 1.0910894 ,  1.52752516]]])
+    b = fe.backend.tensor_normalize(n, 4.5, 2.29128784747792)  # ([[[-1.52752516, -1.0910894 ], [-0.65465364, -0.21821788]], [[ 0.21821788,  0.65465364], [ 1.0910894 ,  1.52752516]]])
+    b = fe.backend.tensor_normalize(n, (4., 5.), (2.23606798, 2.23606798))  # [[[-1.34164073, -1.34164073], [-0.44721358, -0.44721358]], [[ 0.44721358,  0.44721358], [ 1.34164073,  1.34164073]]]
+    ```
+
+    This method can be used with PyTorch tensors:
+    ```python
+    p = torch.tensor([[[1.0, 2.0], [3.0, 4.0]], [[5.0, 6.0], [7.0, 8.0]]])
+    b = fe.backend.tensor_normalize(n, None, None)  # ([[[-1.52752516, -1.0910894 ], [-0.65465364, -0.21821788]], [[ 0.21821788,  0.65465364], [ 1.0910894 ,  1.52752516]]])
+    b = fe.backend.tensor_normalize(n, 4.5, 2.29128784747792)  # ([[[-1.52752516, -1.0910894 ], [-0.65465364, -0.21821788]], [[ 0.21821788,  0.65465364], [ 1.0910894 ,  1.52752516]]])
+    b = fe.backend.tensor_normalize(n, (4., 5.), (2.23606798, 2.23606798))  # [[[-1.34164073, -1.34164073], [-0.44721358, -0.44721358]], [[ 0.44721358,  0.44721358], [ 1.34164073,  1.34164073]]]
+    ```
+
+    Args:
+        tensor: The input value.
+        mean: The mean which needs to applied(eg: None, 3.8, (1.9, 2.0, 2.9))
+        std: The standard deviation which needs to applied(eg: None, 3.8, (1.9, 2.0, 2.9))
+
+    Returns:
+        The normalized values of `tensor`.
+
+    Raises:
+        ValueError: If `tensor` is an unacceptable data type.
+    """
+    channel_level = False
+    if isinstance(mean, Tuple) or isinstance(std, Tuple):
```

Comment:
what if one or other but not both?

Reply:
or if user uses list instead of tuple

Reply:
Currently we do allow if either of them to be None and other not None. The combination would be, (mean = None and std =(0.5, 0.5, 0.5)) or (mean=(0.5, 0.5, 0.5), std=None) or (mean=(0.5, 0.5, 0.5) , std=(0.5,  0.5, 0.5)). In those cases where one of them is None, the other is calculated channel wise mean or std.